include/swift/AST/ASTContext.h

```diff
@@ -338,6 +338,9 @@ class ASTContext final {
       llvm::SmallPtrSet<DerivativeAttr *, 1>>
       DerivativeAttrs;
 
+  llvm::DenseMap<ClassDecl *, llvm::SmallVector<DeclName, 2>>
+      MissingClassVTableEntries;
```

Comment:
Can you store the DeclName in the MissingMemberDecl instead of adding another table to the ASTContext?

Reply:
While possible to iterate the members and pull out the `MissingMemberDecl`, it may limit any additional context from being saved off (due to `Decl`s being bump ptr allocated).

Reply:
Can you explain what you mean by limiting context from being saved off?

What I'm suggesting is iterating over the members in TypeCheckDeclPrimary, yeah.

Reply:
Adding any additional context like the XRefTrace or the module which failed to load would not be possible AIUI.

Reply:
I think you can store that context in the MissingMemberDecl itself (it's fine for bump-allocated Decls to point to other things. You can either bump-allocate the additional context, or heap-allocate it and register a global destructor with the ASTContext on the MissingMemberDecl field in question).

Reply:
SGTM